sim-lib/src/lib.rs

```diff
@@ -155,6 +157,15 @@ impl Display for PaymentResult {
     }
 }
 
+impl Default for PaymentResult {
+    fn default() -> Self {
+        PaymentResult {
+            htlc_count: 0,
+            payment_outcome: PaymentOutcome::Unknown,
```

Comment:
latest revision defines two explicit payment outcomes `NotDispatched` and `TrackPaymentFailed` which we use in case of temporary failures. Related to @sr-gi's comment on not implementing `Default`, we don't need a Default